includes/Modules/Idea_Hub.php

```diff
@@ -770,6 +792,26 @@ private function is_idea_post( $post_id ) {
 		return is_array( $this->get_post_idea( $post_id ) );
 	}
 
+	/**
+	 * Hook to check whether an Idea Hub post status has changed.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string  $new_status Updated post status.
+	 * @param string  $old_status Previous post status.
+	 * @param WP_Post $post The post in question.
+	 */
+	private function on_idea_hub_post_status_transition( $new_status, $old_status, $post ) {
+		if ( ! $this->is_idea_post( $post->ID ) ) {
+			return;
+		}
+
+		if ( $new_status !== $old_status ) {
+			$transients = new Transients( $this->context );
```

Comment:
I see we have many places in the Idea_Hub class where we instantiate new instances of the Transients class. I think it's a good time to refactor it a little bit and add a new property for transients to this class, and then use one instance in all places.